test/compilable/test17419.d

```diff
@@ -35,3 +35,19 @@ void bar()
     void nested() { }
     static assert(__traits(getLinkage, typeof(&nested)) == "D");
 }
+
+class FooD {}
+interface FooDInterface {}
+extern (C++) class FooCpp {}
+extern (C++) interface FooCppInterface {}
+
+static assert(__traits(getLinkage, FooD) == "D");
+static assert(__traits(getLinkage, FooDInterface) == "D");
+static assert(__traits(getLinkage, FooCpp) == "C++");
+static assert(__traits(getLinkage, FooCppInterface) == "C++");
+
+version (D_ObjectiveC)
```

Comment:
Is this guard necessary?  Should this work regardless of whether we're compiling for ObjectiveC or not?

Reply:
Yes it's necessary, this is what happens when compiling an Objective-C interface on Linux: `Error: interface foo.Foo Objective-C interfaces not supported`.